src/api/requests.ts

```diff
@@ -1,60 +1,66 @@
+import { IKartLapsPerRun, ILapDataset } from "../models/go-kart-types";
+
 // Get all sessions
-export default function getAllRuns(callback, errorCallback, finallyCallback) {
+export default function getAllRuns(
+  callback: (allRunsJSON: string[]) => void,
+  errorCallback: (error: Error) => void,
+  finallyCallback: (() => void) | undefined = undefined
+) {
   fetch(`https://go-kart-api.onrender.com/runs`)
     .then((response) => response.json())
-    .then((res) => {
+    .then((res: string[]) => {
       if (!res) {
         errorCallback(new Error("Cannot find Run"));
       } else {
```

Comment:
Else statement not needed after guard clause.

Reply:
@AyaazStreet this is only true if he returns the call to `errorCallback`. As he is only _calling_ it, it does not break out of the `then` context.